modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptFetchClientCodegen.java

```diff
@@ -235,9 +235,24 @@ private void addNpmPackageGeneration() {
         this.updateOperationParameterEnumInformation(operations);
         this.addOperationObjectResponseInformation(operations);
         this.addOperationPrefixParameterInterfacesInformation(operations);
+        this.escapeOperationIds(operations);
         return operations;
     }
 
+    private void escapeOperationIds(Map<String, Object> operations) {
+        Map<String, Object> _operations = (Map<String, Object>) operations.get("operations");
+        List<CodegenOperation> operationList = (List<CodegenOperation>) _operations.get("operation");
+        for (CodegenOperation op : operationList) {
+            String param = op.operationIdCamelCase + "Request";
+            if (op.imports.contains(param)) {
+                // we import a model with the same name as the generated operation, escape it
+                op.operationIdCamelCase += "Operation";
```

Comment:
@wing328 is changing the operation id a behavior a user would expect? or does a user expect the operation id to remain unaltered?

Reply:
For example, if the operationId is a reserved word (e.g. return), the output won't work/compile so we need to escape it.

We're doing the same thing in other generators to ensure the output compiles without issues by escaping operationId, property name, parameter name and more.